src/lib.rs

```diff
@@ -288,7 +324,24 @@ where
     /// assert_eq!(vector, vec![42, 41, 40, 39, 38]);
     /// ```
     pub fn to_vec(&self) -> Vec<Idx> {
-        self.clone().into_iter().collect()
+        self.into_iter().collect()
+    }
+}
+
+impl<'a, Idx> IntoIterator for &'a Range<Idx, Upwards>
+where
+    Idx: Clone + PartialEq + One + AddAssign + SubAssign,
+{
+    type Item = Idx;
+    type IntoIter = RangeIter<Idx, Upwards>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        RangeIter {
+            current: self.from.clone(),
```

Comment:
Looking at this here, it might be a good idea to mention in the docs of .iter() (as well as in a new doc comment for this impl) that the index will be cloned. I still have no concrete example, but in theory an index could use reference counting and internal mutability (for whatever reason).